### DISCUSSION TITLE
Home-brew Support

### DISCUSSION THREAD

**Discussion by derpycoder** [authelia/authelia#4621] [Ideas]
### Description

I would like an easy way to download and setup Authelia on Mac, so that I can learn to setup the Auth locally as well as have dev & prod parity.

### Use Case

I am running all the stuff that is going to run in production locally, so that I can learn and configure everything that will go to prod with ease.

For example, I have several production features enabled in local:
1. HTTPS, WSS
2. Custom Domain using etc/hosts
3. HTTP/2
4. Compression
5. Secure Cookies
6. Several other companion services(HAProxy, Meilisearch, CockroachDB, PGWeb, MinIO, Imgproxy, Varnish Cache). 
7. TLS Certs, Mkcert, Salting

All of the above help me figure out how things work, and allows me to learn at my own pace. It also helped me learn how Apps with TLS behave and communicate.

And even though I have Authentication & Authorization in my application, I would like a cross app proxy Auth companion for all the subdomains!!

### Details

I just want to do:

`brew install authelia` or specify the version in asdf's `.tool-versions` file.

### Documentation
Example:
[Install FusionAuth using Homebrew](https://fusionauth.io/docs/v1/tech/installation-guide/homebrew)

**Comment by derpycoder** [comment]
My `.tool-versions` file:
| Tool      | Versions             |
|-----------|----------------------|
| mc        | 2022-09-16T09-16-47Z |
| minio     | 2022-09-17T00-09-45Z |
| elixir    | 1.13.4-otp-25        |
| erlang    | 25.0.3               |
| nodejs    | 18.9.1               |
| cockroach | 22.1.6               |

My `Taskfile.yaml`
| Cmds                               |
|----------------------------------------|
| asdf install                           |
| brew install pgweb                     |
| brew install mkcert                    |
| brew install netdata                   |
| brew install varnish                   |
| brew install haproxy                   |
| brew install imgproxy                  |
| brew install meilisearch               |
| brew install kevinburke/safe/hostsfile |
| mix local.hex                          |

**Comment by nightah** [comment]
We used to provide Darwin binaries, see #967 and #985. This was removed because there really didn't seem to be much interest in running Authelia directly on OSX.

I'm happy to leave this issue open to be voted as an FR but unless there's significant interest we likely will not prioritise bringing this back including a Brew formula.

**Comment by derpycoder** [comment]
I see.

Thanks for getting back to me. (I thought because fusion auth provides a brew tap, Authelia might consider it. 😅)

I suppose there won't be any interest for this in the future as everyone just went along with Docker.

I prefer baremetal over another layer of abstraction.

Is it possible to get a single go binary running?

Also, what's the closest thing to Authelia in terms of performance and memory usage?!?

Authentik & Fusion Auth are a no no as they are slow, and Ory is kind of DIY.

**Comment by nightah** [comment]
You can actually build the binary on OSX quite simply yourself with Go and npm/yarn.

I'm not completely averse to providing a Darwin binary again. That's not too much work we just need to update some of our build tooling.

I haven't actually looked at how difficult or easy it would be to create a brew formula. The main issue with creating that is that none of the team really uses OSX which makes supporting it immensely difficult.

If you would be happy with a binary that you can run yourself on bare metal I can discuss with the team to bring this back.

**Comment by derpycoder** [comment]
I will love if there's a binary for mac.

Also I was going to try out creation of a Homebrew Formula.

[Hostsfile Homebrew Formula - Kevin Burke](https://github.com/kevinburke/homebrew-safe/blob/main/Formula/hostsfile.rb)

[Make your go binaries Homebrew Installable - Kevin Burke](https://kevin.burke.dev/kevin/install-homebrew-go/)

I will learn it tonight.

**Comment by derpycoder** [comment]
@nightah,

Checkout [GoReleaser](https://github.com/goreleaser/goreleaser)

GoReleaser builds Go binaries for several platforms, creates a Github release and then pushes a Homebrew formula to a tap repository.

**Comment by derpycoder** [comment]
I failed to make it work with Homebrew. 😅

**Comment by james-d-elliott** [comment]
Thanks for submitting this feature request and making an attempt to solve it (if you jump on chat we may be able to solve your issues with getting it to work). We've decided to for the time being convert this issue to an idea discussion for the following reasons:

- This idea has __*NOT*__ been rejected/declined/etc.
- This prevents our issue list being obscure due to excessive items that we have no direct plans to implement at this stage.
- This prevents this reasonable idea from being lost if it were closed.
- This allows people to vote on the idea.

**Comment by Mactastic1-5** [comment]
I want to get it working on Mac.

**Reply by james-d-elliott** [reply]
We would welcome the PR.

**Reply by Mactastic1-5** [reply]
Didn't it not run correctly?

**Reply by james-d-elliott** [reply]
They couldn't get it working and provided no context. As to what actual issue there was I have no clue. You said you want to get it working for a specific use case, I'm saying we'd welcome a PR if you get that use case working.

**Reply by nightah** [reply]
> Didn't it not run correctly?

We have had working OSX binaries in the past, we never invested any time in looking at Brew.
